extends Control





var tutorial_activo = true

func _ready():
 
    iniciar_dialogo_espacio()
    
    
func iniciar_dialogo_espacio():
    var gato_dialogo = preload("res://escenas/Gato_Instrucciones/Gato_Instrucciones.tscn").instantiate()
    add_child(gato_dialogo)
    
    gato_dialogo.global_position = Vector2(10, 500)
    
    # Textura que se ve mientras habla
    gato_dialogo.textura_habla_personal = preload("res://assets/sprites/ui/Espacio/Galileo_Espacio_Hablando.png")
    # Textura de reposo cuando no habla
    gato_dialogo.textura_idle_personal = preload("res://assets/sprites/ui/Espacio/Galileo_Espacio_Base.png")
    
    # Ajustar escala del Sprite segÃºn la textura
    var tamaÃ±o_deseado = Vector2(256,256)  # el tamaÃ±o que quieres en tu escena
    var tamaÃ±o_textura = gato_dialogo.textura_habla_personal.get_size()
    gato_dialogo.gato.scale = tamaÃ±o_deseado / tamaÃ±o_textura
    
    # DiÃ¡logos que quieres que diga en este escenario
    gato_dialogo.dialogos = [
    "Â¡Hola, astronauta! ğŸš€ Hoy viajamos al espacio exterior.",
    "Mira esas estrellas brillantes...",
    "Tu misiÃ³n serÃ¡ tomar fotos de los planetas. ğŸ“¸",
    "Algunos se mueven rÃ¡pido, Â¡asÃ­ que apunta bien! ğŸ›°ï¸",
    "Cuando termines todas las fotos, tendrÃ¡s una sorpresa estelar. ğŸŒ "
]


    # âœ… Esperar hasta que termine el diÃ¡logo
    await gato_dialogo.dialogo_terminado

    # âœ… Ahora sÃ­: empezar a mostrar animales
    tutorial_activo = false
